strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/NpmLoginAuthenticationSupplier.java

```diff
@@ -0,0 +1,83 @@
+package org.carlspring.strongbox.security.authentication.suppliers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.carlspring.strongbox.authentication.api.password.PasswordAuthentication;
+import org.carlspring.strongbox.controllers.layout.npm.NpmArtifactController;
+import org.carlspring.strongbox.controllers.layout.npm.NpmUser;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+@Order(4)
```

Comment:
it probably should work with `@Order(3)`, same as for `NugetApiKeyAuthenticationSupplier`. in this case we can leave `BasicAuthenticationSupplier` with `@Order(4)`.

Reply:
ok so change it to order 3 and BasicAuth back to Order 4

Reply:
inheritance is good here, it would be great if you can do it!
it probably will inherit the `@Order` as well

Reply:
this is regarding above comment :)